src/metabase/models/params/card_values.clj

```diff
@@ -0,0 +1,52 @@
+(ns metabase.models.params.card-values
+  "Code related to getting values for a parameter where its source values is a card."
+  (:require
+    [clojure.string :as str]
+    [metabase.models :refer [Card]]
+    [metabase.query-processor :as qp]
+    [metabase.util.schema :as su]
+    [schema.core :as s]
+    [toucan.db :as db]))
+
+(def ^:dynamic *max-rows*
+  "Maximum number of rows returned when running a card."
+  2000)
+
+(defn- values-from-card-query
+  [card-id value-field query]
+  (let [card        (db/select-one Card :id card-id)]
+    {:database (:database_id card)
+     :type     :query
+     :query    (merge
+                 {:source-table (format "card__%d" card-id)
+                  :fields       [value-field]
+                  :limit        *max-rows*}
+                 (when query
+                   {:filter [:contains [:lower value-field] (str/lower-case query)]}))
+     :middleware {:disable-remaps? true}}))
```

Comment:
Last question - why is `:disable-remaps?` necessary? I don't see a test for it.

Reply:
https://www.metabase.com/docs/latest/data-modeling/metadata-editing.html#remapping-column-values

We have a feature to remap fk to a field.
This is done in `add-dimension-projections` middleware.

For values for parameter, we want to disable this feature because of cases where you select an FK field as a values source.
And if remapping is enabled the returned value will not be the `fk` field but the field it's mapped to.

Anw, good catch I'll add a test for it.

Reply:
added here https://github.com/metabase/metabase/pull/27343/commits/939983670dda1fdd1b01056db144ea6257a246af

Reply:
> Actually, if we want to go far we could still display the remapped values and swap it with real values when applying in a filter. But that's too much effort for a niche feature IMO.

Yeah I think that's fair, and the use case is exactly what the labels should cover. Thanks